### PR TITLE
Handle non-ascii output from running commands in python wrapper

### PR DESCRIPTION
**Pull request by simathih**
fix for https://portal.microsofticm.com/imp/v5/incidents/details/536301475/summary

### CODE REVIEW

**Comment by hestolz** [comment]
is this verified on python2 and python3 both?

**Comment by simathih** [comment]
> is this verified on python2 and python3 both?

yeah verified in python 2 and 3:
https://onecompiler.com/python2/42qmmxpjm
https://onecompiler.com/python/42qmmw9eu

File: AzureMonitorAgent/agent.py [hunk 1]
```diff
@@ -2229,7 +2229,7 @@ def run_get_output(cmd, chk_err = False, log_cmd = True):
             exit_code = e.returncode
             output = e.output
 
-    if type(output) == str:
+    if all(ord(c) < 128 for c in output):
```
Comment (hestolz) [inline AzureMonitorAgent/agent.py:2232]: what about doing `errors='replace'` instead of checking the input?

Comment (simathih) [reply AzureMonitorAgent/agent.py:2232]: this works in python3 but not python2:
 txt = txt.encode("utf-8", errors='replace')
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 13: ordinal not in range(128)

https://onecompiler.com/python2/42qmtwtm5

**Review by hestolz — APPROVED** [review]

**Pull request merged**